### DISCUSSION TITLE
Using constr (or conbytes, etc) raises mypy errors

### DISCUSSION THREAD

**Discussion by dantebben** [pydantic/pydantic#2571] [Question] [answered]
I would like to be able to reuse constr types for multiple attributes, for multiple models.  The code works as expected, and creates a JSON schema as expected, but mypy raises errors.

I am using:
- Python = 3.9.1
- mypy = 0.812
- pydantic =1.8.1

My mypi.ini looks like:
```ini
[mypy]
plugins = pydantic.mypy
```

Example Code
```python
from __future__ import annotations

from pydantic import BaseModel, constr


# Desire is to reuse constr types for multiple attributes
RegexStr = constr(regex="^(Hello|World)$")


class MyModel(BaseModel):
    str1: RegexStr  # mypy raises 'not valid as a type'
    str2: constr(regex="^(Hello|World)$")  # mypy raises 'Invalid type comment or annotation
    val1: int


# This will validate correctly
my_model = MyModel(str1="Hello", str2="World", val1=10)
print(my_model)

# This will correctly raise two ValidationErrors
my_wrong_model = MyModel(str1="Aloha", str2="Earth", val1=5)
```
The output of mypy is:
```bash

$ mypy my_model.py
my_model.py:11: error: Variable "my_model.RegexStr" is not valid as a type
my_model.py:11: note: See https://mypy.readthedocs.io/en/latest/common_issues.html#variables-vs-type-aliases
my_model.py:12: error: Invalid type comment or annotation
my_model.py:12: note: Suggestion: use constr[...] instead of constr(...)
Found 2 errors in 1 file (checked 1 source file)

```

Am I doing something wrong?  Is this the expected behavior, and if so, is there a workaround?  Is this a bug?

Note, mypy has similar error messages when run against the example [here](https://pydantic-docs.helpmanual.io/usage/types/#constrained-types)

**Comment by PrettyWood** [comment, marked as answer]
Hi @dantebben 
You're doing it right and it's a known issue.
The best way now is probably to write
```py
from typing import Annotated

from pydantic import BaseModel, Field

class MyModel(BaseModel):
    str1: Annotated[str, Field(regex="^(Hello|World)$")]
    str2: Annotated[str, Field(regex="^(Hello|World)$")]
    val1: int
```
This is the easiest way to be compliant with `mypy`

**Reply by dantebben** [reply]
@PrettyWood 
Awesome, thanks.  I think this will work for me.

It even works to define the alias as 
```python
RegexStr = Annotated[str, Field(regex="^(Hello|World)$")]
```
To be able to reuse it in multiple models and attributes.
